From e702a9a0c9fd3e7cb55780cd107d8cc639da020f Mon Sep 17 00:00:00 2001
From: li zhuangzhi <zhuangzhi.li@intel.com>
Date: Wed, 4 Sep 2019 15:09:50 +0800
Subject: [PATCH 4/4] HDCP will not be started until HWC is ready

* If HDCP started before HWC, HWC will not get DRM master.
so only start HDCP when HWC is ready
* Clean up invalid configurations

Change-Id: Ie615b925bd35c8678a9b867d34192337dccc2edd
Tracked-On:
Signed-off-by: li zhuangzhi <zhuangzhi.li@intel.com>
---
 groups/graphics/auto/init.rc.1 | 17 ++++-------------
 1 file changed, 4 insertions(+), 13 deletions(-)

diff --git a/groups/graphics/auto/init.rc.1 b/groups/graphics/auto/init.rc.1
index adb5a29..6adb52c 100644
--- a/groups/graphics/auto/init.rc.1
+++ b/groups/graphics/auto/init.rc.1
@@ -14,12 +14,15 @@ service coreu /system/vendor/bin/coreu
 # and once that is set up it will drop all unnecessary capabilities and
 # will not show up as a root process in the steady state.
 service hdcpd /system/vendor/bin/hdcpd
-    class main
+    disabled
     user media
     group media shell
     seclabel u:r:hdcpd:s0
 {{/coreu_msync}}
 
+on property:vendor.hwc.ready=1
+    start hdcpd
+
 on post-fs-data
     # Change ownership of pci syfs file to 'media', as hdcpd runs as 'media'
     chown media media /sys/devices/pci0000\:00/0000\:00\:02.0/resource0
@@ -43,15 +46,3 @@ on boot
     chown media media /sys/kernel/debug/tracing/events/i915/i915_ring_wait_end/enable
     chown media media /sys/devices/pci0000:00/0000:00:02.0/drm/card0/gt_max_freq_mhz
     chown media media /sys/devices/pci0000:00/0000:00:02.0/drm/card0/gt_min_freq_mhz
-
-on property:persist.vendor.gen_gfxd.enable=1
-    start gfxd
-
-on property:persist.vendor.gen_gfxd.enable=0
-    stop gfxd
-
-service gfxd /system/vendor/bin/gfxd
-    class main
-    user root
-    group graphics
-    disabled
-- 
2.7.4

